iPetCare/Application/Services/InvitationService.cs

```diff
@@ -87,71 +87,148 @@ public async Task<ServiceResponse> DeleteInvitationAsync(Guid invitationId)
             return new ServiceResponse(HttpStatusCode.OK);
         }
 
-        public async Task<ServiceResponse<ChangeStatusInvitationDtoResponse>> ChangeInvitationStatusAsync(ChangeStatusInvitationDtoRequest dto, Guid invitationId)
+        public async Task<ServiceResponse> AcceptInvitationAsync(Guid invitationId)
         {
             if (CurrentlyLoggedUser == null)
-                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.Unauthorized);
+                return new ServiceResponse(HttpStatusCode.Unauthorized);
 
             var invitation = await Context.Requests.FindAsync(invitationId);
 
-            if (invitation == null)
-                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.NotFound);
+            if(invitation == null)
+                return new ServiceResponse(HttpStatusCode.NotFound);
 
-            var pet = await Context.Pets.FindAsync(invitation.PetId);
+            if(CurrentlyLoggedUser.Id != invitation.Pet.OwnerPets.FirstOrDefault(x => x.MainOwner)?.Owner.UserId)
+                return new ServiceResponse(HttpStatusCode.Forbidden);
 
-            if (CurrentlyLoggedUser.Id == invitation.UserId && pet.OwnerPets.Where(x => x.Owner.User.Id == CurrentlyLoggedUser.Id).Any())
-                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.Forbidden);
+            var user = await UserManager.FindByIdAsync(invitation.UserId);
 
-            invitation.IsAccepted = dto.IsAccepted;
+            if(user == null)
+                return new ServiceResponse(HttpStatusCode.NotFound);
 
-            var responseDto = Mapper.Map<ChangeStatusInvitationDtoResponse>(invitation);
+            if (user.Role == Role.Vet)
+            {
+                var vetPet = new VetPet()
+                {
+                    Vet = user.Vet,
+                    Pet = invitation.Pet,
+                };
 
-            if (dto.IsAccepted)
+                await Context.VetPets.AddAsync(vetPet);
+            }
+            else if (user.Role == Role.Owner)
             {
-                var User = UserManager.FindByIdAsync(invitation.UserId).Result;
+                var ownerPet = new OwnerPet()
+                {
+                    Owner = user.Owner,
+                    Pet = invitation.Pet,
+                    MainOwner = false,
+                };
+                await Context.OwnerPets.AddAsync(ownerPet);
+            }
+
+            Context.Requests.Remove(invitation);
+
+            int result = await Context.SaveChangesAsync();
+
+            return result >= 0
```

Comment:
Na pewno >= 0? 
Tzn wydaje mi się, że saveChanges zwróci 0 - czyli nic się nie zmieniło - to wcale nie powinna być okejka

Czy powinna być?